src/build/__init__.py

```diff
@@ -49,49 +49,47 @@ class TypoWarning(Warning):
     """
 
 
-class IncompleteCheckWarning(Warning):
+def check_dependency(req_string, ancestral_req_strings=(), parent_extras=frozenset()):
```

Comment:
I think it would be better to put effort in upstreaming this into pypa/packaging first, as proposed in pypa/packaging#317.

Reply:
The impression I got was that they were reluctant to add anything which manipulates distributions.  `distlib` would probably be a good fit but everybody seems to be moving away from it.

Reply:
In my experience there's a long cycle with adding stuff to distlib... so if we want it soon we better add it here. We can always remove it if we do manage to upstream it.

Reply:
I am in no hurry to merge this as it only affects people using `--no-isolation` when trying to verify recursive extras (extras that depend on extras), which is mostly distro packagers, for whom this is only a nice to have. I would rather put the effort into merging somewhere this could live long term rather than here. Before merging it here, we should figure out what would it take to put it somewhere else and then make the call.

Reply:
> I am in no hurry to merge this as it only affects people using `--no-isolation`

As a (new) contributor to this project, this sentence on its own would be enough for me to abandon spending any more effort on this PR.

Reply:
> as it only affects people ... trying to verify recursive extras

`check_version` would not verify transitive dependencies at all, whether or not they came from extras.

Reply:
Yes sorry, my bad.